core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraFromRjsAdapter.kt

```diff
@@ -49,63 +68,498 @@ import fr.sncf.osrd.utils.distanceRangeMapOf
+        if ((currentCurve.upper == upper) and currentCurveIt.hasNext()) {
+            currentCurve = currentCurveIt.next()
+        }
+        lower = upper
+    }
+
+    return gradients
+}
+
+/** Builds the ranges of blocked loading gauge types on the track */
+private fun getBlockedGauge(rjsSection: RJSTrackSection): DistanceRangeMap<LoadingGaugeConstraint> {
+    // This method has a bad complexity compared to more advanced solutions,
+    // but we don't expect more than a few ranges per section.
+    // TODO: use an interval tree
+    val res = distanceRangeMapOf<LoadingGaugeConstraint>()
+    if ((rjsSection.loadingGaugeLimits == null) || (rjsSection.loadingGaugeLimits.isEmpty()))
+        return res
+
+    // Sorts and removes duplicates
+    val transitions = TreeSet<Double>()
+    for (range in rjsSection.loadingGaugeLimits) {
+        transitions.add(range.begin)
+        transitions.add(range.end)
+    }
+
+    val transitionsList = ArrayList(transitions) // Needed for index based loop
+    for (i in 1 until transitionsList.size) {
+        val begin = transitionsList[i - 1]
+        val end = transitionsList[i]
+        val allowedTypes = HashSet<RJSLoadingGaugeType>()
+        for (range in rjsSection.loadingGaugeLimits) {
+            if (range.begin <= begin && range.end >= end) {
+                val compatibleTypes = compatibleGaugeTypeMap[range.category]
+                if (compatibleTypes == null) {
+                    logger.warn("Invalid gauge type ${range.category} for track ${rjsSection.id}")
+                    allowedTypes.addAll(allLoadingGaugeTypeSet)
+                } else {
+                    allowedTypes.addAll(compatibleTypes)
+                }
+            }
+        }
+        val blockedTypes = Sets.difference(enumValues<RJSLoadingGaugeType>().toSet(), allowedTypes)
+        res.put(
+            begin.meters,
+            end.meters,
+            LoadingGaugeConstraintImpl(ImmutableSet.copyOf(blockedTypes))
+        )
+    }
+    return res
+}
+
+private fun buildCompatibleGaugeTypesMap(): Map<RJSLoadingGaugeType, Set<RJSLoadingGaugeType>> {
+    val compatibleGaugeType = hashMapOf<RJSLoadingGaugeType, Set<RJSLoadingGaugeType>>()
+    for (gaugeType in enumValues<RJSLoadingGaugeType>()) {
+        val compatibleTypes = gaugeType.compatibleGaugeTypes
+        if (compatibleTypes != null) compatibleGaugeType[gaugeType] = compatibleTypes
+    }
+    return compatibleGaugeType
+}
+
+@JvmField val compatibleGaugeTypeMap = buildCompatibleGaugeTypesMap()
+@JvmField val allLoadingGaugeTypeSet = enumValues<RJSLoadingGaugeType>().toSet()
+
+private fun parseRjsRouteWaypoint(
+    rjsWaypoint: RJSRouteWaypoint,
+    builder: RawInfraFromRjsBuilderImpl,
+    sectionNameToOffsetToDetectorId: MutableMap<String, TreeSet<Offset<TrackSection>>>,
+    detectorMap: MutableMap<String, DetectorId>,
+) {
+    detectorMap[rjsWaypoint.id] = builder.detector(rjsWaypoint.id)
+
+    if (rjsWaypoint.position == 0.0) return
```

Comment:
The detectors at offset 0 are ignored because the map is only used to build chunks, right?

I think a comment would help here, the map doesn't contain what I'd expect from its name. We could also add them anyway and then filter them out when building the chunks.

Reply:
Indeed, the name was a leftover from previous ideas, renamed and commented in https://github.com/osrd-project/osrd/pull/6722/commits/d47d21a0d5092e3184dfafbb79bc63d946fb12ed